dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1079,6 +1130,11 @@ class PerfTest {
       final String? localEngine = localEngineFromEnv;
       final String? localEngineSrcPath = localEngineSrcPathFromEnv;
 
+      if (forceOpenGLES ?? false) {
+        assert(enableImpeller!);
+        _addOpenGLESToManifest(testDirectory);
```

Comment:
This is probably overly paranoid, but I think restoring the file to its original contents in a `finally {}` couldn't hurt.

Reply:
If there is an error inside of `_addOpenGLESToManifest` there is no need to restore the file since the file is only edited in the last line.

It looks like `writeAsStringSync` is not atomic so if writing somehow failed it could be garbage.  But the cases where the writing fail would probably be an environment where the test wouldn't execute properly anyways, no?

This test is modifying its own copy of the gallery in its own temporary directory so this shouldn't effect other tests, see https://github.com/flutter/flutter/pull/131796/files#diff-514ad19bc7a9b9e196b1568127fed85ee10552d62baa32f960d694a98cfc83f1R16.

Let me know if there is a situation I'm not considering.

Reply:
Like I said, I'm maybe being a bit paranoid, but suppose `_addOpenGLESToManifest` is used by mistake for an some other integration test that lives in `flutter/flutter`, and that the change to the xml file persists by mistake across multiple benchmark runs.

Reply:
Okay, done.